front/app/containers/Admin/projects/project/index.test.tsx

```diff
@@ -183,205 +201,240 @@ describe.skip('<AdminProjectEdition />', () => {
     const phases = [mockPhaseInformationData, mockPhaseSurveyTypeformData];
     const routerProps = getRouterProps('timelineIdeation');
 
-    // passing in mock data to replace the HoCs
-    // ignore the error, shallowWithIntl method passes in the intl object
-    const wrapper = shallowWithIntl(
+    render(
       <AdminProjectsProjectIndex
         surveys_enabled={surveys_enabled}
         typeform_enabled={typeform_enabled}
         phases={phases}
-        granularPermissionsEnabled={granularPermissionsEnabled}
         project={project}
-        children={children}
+        intl={intl}
         {...routerProps}
         {...localizeProps}
-      />
+        {...additionalProps}
+      >
+        {children}
+      </AdminProjectsProjectIndex>
     );
-    const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;
 
-    expect(tabs.map((tab) => tab.url)).toMatchSnapshot();
+    const tabNames = [
+      'General',
+      'Description',
+      'Input manager',
+      'Survey Results',
+      'Timeline',
+      'Events',
+    ];
+    tabNames.forEach((tabName) => {
+      expect(screen.getByText(tabName)).toBeInTheDocument();
+    });
+    expect(screen.getAllByTestId('e2e-resource-single-tab')).toHaveLength(
+      tabNames.length
+    );
   });
 
-  it('renders the correct set of tabs for a timeline project with a googleform survey phase', () => {
+  // this one seems to oscillate between 5 and 7 tabs...
+  it('renders the correct set of tabs for a timeline project with a googleform survey phase', async () => {
     const surveys_enabled = true;
     const typeform_enabled = true;
-    const granularPermissionsEnabled = true;
 
     const project = getProject('timelineIdeation', 'timeline');
     const phases = [mockPhaseInformationData, mockPhaseSurveyGoogleFormData];
     const routerProps = getRouterProps('timelineIdeation');
 
-    // passing in mock data to replace the HoCs
-    // ignore the error, shallowWithIntl method passes in the intl object
-    const wrapper = shallowWithIntl(
+    render(
       <AdminProjectsProjectIndex
         surveys_enabled={surveys_enabled}
         typeform_enabled={typeform_enabled}
         phases={phases}
-        granularPermissionsEnabled={granularPermissionsEnabled}
         project={project}
-        children={children}
+        intl={intl}
         {...routerProps}
         {...localizeProps}
-      />
+        {...additionalProps}
+      >
+        {children}
+      </AdminProjectsProjectIndex>
     );
-    const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;
 
-    expect(tabs.map((tab) => tab.url)).toMatchSnapshot();
+    const tabNames = [
+      'General',
+      'Description',
+      'Input manager',
+      'Timeline',
+      'Events',
+    ];
+    tabNames.forEach((tabName) => {
+      expect(screen.getByText(tabName)).toBeInTheDocument();
+    });
+    expect(screen.getAllByTestId('e2e-resource-single-tab')).toHaveLength(
```

Comment:
Isn't it a violation to have the same (test) id for multiple tabs?

Reply:
I don't think so, that `getAllByTestId` returns multiple instances of the same test-id. The alternative would be adding a classname but I think that implies styling, this way it's explicit that it's been added for testing purposes